internal/install/recipes/service_recipe_fetcher.go

```diff
@@ -77,7 +77,29 @@ func (f *ServiceRecipeFetcher) FetchRecommendations(ctx context.Context, manifes
 		return nil, err
 	}
 
-	return resp.Docs.OpenInstallation.Recommendations.ToRecipes(), nil
+	allRecipes := resp.Docs.OpenInstallation.Recommendations.ToRecipes()
```

Comment:
Do we need this dedup logic?
Normally the searchRecipe and process detection should match to subset of recipes, and when the recommendations are fetched it should be leveraging the process match to only pick 1 recipe for a given name.

Maybe there is another change coming, but another key part is to add the os/platform criteria when we search for recipes. For example we should only return recipes that are meant for the current os and/or linux version. We wouldn't want to search and returns recipes that are not meant to be installed on the current host os/platform.

Reply:
I believe this was mostly a guard against the service sending back two recipes with the same name.  This was the case in some circumstances that caused the recipes to be installed more than once in a single run.

I don't think this affects the process filtering currently, but are you suggesting an additional change to the process filter?

Reply:
Got it, maybe this is needed I'm unsure. I dont quite like that we arbitrarily pick one, maybe we picked the wrong one?

Yeah, I think we need another change that might prevent the case of having duplicate named recipes coming back.
For example, if I'm running on an aws linux2 os/platform, I shouldn't get back a searched recipe that targets debian, or even more so a recipe targeting windows.